repair/repair.hh

```diff
@@ -147,6 +148,7 @@ public:
         : all(std::move(a))
         , mandatory(std::move(m)) {
     }
+    repair_neighbors(std::vector<gms::inet_address> nodes, std::vector<shard_id> shards);
```

Comment:
It's a bit weak that we can use the wrong constructor and get an instance of repair_neighbors without shards. But we'll have to live with it.

Reply:
The vnode repair still uses the repair_neighbors without shards.